core/functions.py

```diff
@@ -36,46 +37,87 @@ def init(path=None, folder=None, config_settings:Dict=None, quiet=False, use_exi
+        self.otherJM = otherJM
 
-    config.update(config_settings or {})
+        if config.init:
+            return
 
-    from .jobmanager import JobManager
-    config.default_jobmanager = JobManager(config.jobmanager, path, folder, use_existing_folder)
+        if 'PLAMSDEFAULTS' in os.environ and isfile(expandvars('$PLAMSDEFAULTS')):
+            defaults = expandvars('$PLAMSDEFAULTS')
+        elif 'AMSHOME' in os.environ and isfile(opj(expandvars('$AMSHOME'), 'scripting', 'scm', 'plams', 'plams_defaults')):
+            defaults = opj(expandvars('$AMSHOME'), 'scripting', 'scm', 'plams', 'plams_defaults')
+        else:
+            defaults = opj(dirname(dirname(__file__)), 'plams_defaults')
+            if not isfile(defaults):
+                raise PlamsError('plams_defaults not found, please set PLAMSDEFAULTS or AMSHOME in your environment')
+        with open(defaults, 'r') as f:
+            exec(compile(f.read(), defaults, 'exec'))
 
-    if not quiet:
-        log('Running PLAMS located in {}'.format(dirname(dirname(__file__))), 5)
-        log('Using Python {}.{}.{} located in {}'.format(*sys.version_info[:3], sys.executable), 5)
-        log('PLAMS defaults were loaded from {}'.format(defaults), 5)
+        config.update(config_settings or {})
 
-        log('PLAMS environment initialized', 5)
-        log('PLAMS working folder: {}'.format(config.default_jobmanager.workdir), 1)
+        from .jobmanager import JobManager
+        config.default_jobmanager = JobManager(config.jobmanager, path, folder, use_existing_folder)
 
-    config.slurm = _init_slurm() if "SLURM_JOB_ID" in os.environ else None
+        if not quiet:
+            log('Running PLAMS located in {}'.format(dirname(dirname(__file__))), 5)
+            log('Using Python {}.{}.{} located in {}'.format(*sys.version_info[:3], sys.executable), 5)
+            log('PLAMS defaults were loaded from {}'.format(defaults), 5)
 
-    try:
-        import dill
-    except ImportError:
-        log('WARNING: importing dill package failed. Falling back to the default pickle module. Expect problems with pickling', 1)
+            log('PLAMS environment initialized', 5)
+            log('PLAMS working folder: {}'.format(config.default_jobmanager.workdir), 1)
+
+        config.slurm = _init_slurm() if "SLURM_JOB_ID" in os.environ else None
+
+        try:
+            import dill
+        except ImportError:
+            log('WARNING: importing dill package failed. Falling back to the default pickle module. Expect problems with pickling', 1)
+
+        config.init = True
 
-    config.init = True
+    def __enter__(self):
+        """Enter the context manager; return |init|."""
+        return self
+
+    def __exit__(self, exc_type, exc_value, traceback):
+        """Exit the context manager; call |finish|."""
+        finish(self.otherJM)
+
+    def __repr__(self):
```

Comment:
What is the purpose of having this defined?

Reply:
See https://github.com/SCM-NV/PLAMS/pull/112#discussion_r1232192760